Skip ProcessGroupGlooAyncTest if there is no CUDA available

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30345 Skip ProcessGroupGlooAyncTest if there is no CUDA available**

Skip ProcessGroupGlooAyncTest if there is no CUDA available, otherwise in sandcastle non GPU host the test will abort with failing to load CUDA library

Differential Revision: [D18665322](https://our.internmc.facebook.com/intern/diff/D18665322/)

zhaojuanmao committed:
Skip ProcessGroupGlooAyncTest if there is no CUDA available

Skip ProcessGroupGlooAyncTest if there is no CUDA available, otherwise in sandcastle non GPU host the test will abort with failing to load CUDA library

Differential Revision: [D18665322](https://our.internmc.facebook.com/intern/diff/D18665322/)

[ghstack-poisoned]

rohan-varma commented on torch/lib/c10d/test/ProcessGroupGlooAsyncTest.cpp:
```diff
@@ -244,6 +244,10 @@ void runAsyncBroadcastTest(
 }
 
 int main(int argc, char** argv) {
+  if (cudaNumDevices() == 0) {
```
nit:you can also use `(!at::cuda::is_available())`

rohan-varma commented on torch/lib/c10d/test/ProcessGroupGlooAsyncTest.cpp:
```diff
@@ -244,6 +244,10 @@ void runAsyncBroadcastTest(
 
 int main(int argc, char** argv) {
+  if (cudaNumDevices() == 0) {
+    std::cout << "Skipping test since CUDA is not available" << std::endl;
```
nit:`LOG(INFO) << ...`

rohan-varma reviewed (approved):
Thanks for improving our test reliability!

zhaojuanmao committed:
Update on "Skip ProcessGroupGlooAyncTest if there is no CUDA available"


Skip ProcessGroupGlooAyncTest if there is no CUDA available, otherwise in sandcastle non GPU host the test will abort with failing to load CUDA library

Differential Revision: [D18665322](https://our.internmc.facebook.com/intern/diff/D18665322/)

[ghstack-poisoned]
